huaweicloud/services/apig/resource_huaweicloud_apig_instance_feature.go

```diff
@@ -0,0 +1,262 @@
+// GetInstanceFeature is a method that used to query the list of the features under specified APIG instance.
+func GetInstanceFeature(client *golangsdk.ServiceClient, instanceId, featureName string) (interface{}, error) {
+	httpUrl := "v2/{project_id}/apigw/instances/{instance_id}/features?limit=500"
+	listPath := client.Endpoint + httpUrl
+	listPath = strings.ReplaceAll(listPath, "{project_id}", client.ProjectID)
+	listPath = strings.ReplaceAll(listPath, "{instance_id}", instanceId)
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	offset := 0
+	var feature interface{}
+	for {
+		listPathWithOffset := fmt.Sprintf("%s&offset=%d", listPath, offset)
+		requestResp, err := client.Request("GET", listPathWithOffset, &opt)
+		if err != nil {
+			return nil, err
+		}
+
+		respBody, err := utils.FlattenResponse(requestResp)
+		if err != nil {
+			return nil, err
+		}
+		features := utils.PathSearch("features", respBody, make([]interface{}, 0)).([]interface{})
+		if len(features) < 1 {
+			break
+		}
+
+		if feature = utils.PathSearch(fmt.Sprintf("[?name=='%s']|[0]", featureName), features, nil); feature != nil {
+			return feature, nil
+		}
+		offset += len(features)
+	}
+	return nil, golangsdk.ErrDefault404{}
+}
+
+func resourceInstanceFeatureRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg        = meta.(*config.Config)
+		region     = cfg.GetRegion(d)
+		instanceId = d.Get("instance_id").(string)
+	)
+	client, err := cfg.NewServiceClient("apig", region)
+	if err != nil {
+		return diag.Errorf("error creating APIG Client: %s", err)
+	}
+
+	feature, err := GetInstanceFeature(client, instanceId, d.Id())
+	if err != nil {
+		// When instance ID not exist, status code is 404, error code id APIG.3030
+		return common.CheckDeletedDiag(d, err, "Instance feature configuration")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("name", utils.PathSearch("name", feature, nil)),
+		d.Set("enabled", utils.PathSearch("enable", feature, false)),
+		d.Set("config", utils.PathSearch("config", feature, nil)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceInstanceFeatureUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg         = meta.(*config.Config)
+		httpUrl     = "v2/{project_id}/apigw/instances/{instance_id}/features"
+		instanceId  = d.Get("instance_id").(string)
+		featureName = d.Get("name").(string)
+	)
+	client, err := cfg.ApigV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG client: %s", err)
+	}
+
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	path = strings.ReplaceAll(path, "{instance_id}", instanceId)
+	opts := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildConfigInstanceFeatureParams(featureName, d.Get("enabled"), d.Get("config"))),
+	}
+	_, err = client.Request("POST", path, &opts)
```

Comment:
Shall we need to provide the retry function as the creation logic?

Reply:
yes, the corresponding logic has been added.